src/pages/workspaces/workspace/ShareWorkspaceModal.js

```diff
@@ -74,93 +74,52 @@ const AclInput = ({ value, onChange, disabled, maxAccessLevel, autoFocus }) => {
   ])
 }
 
-export default Utils.withCancellationSignal(class ShareWorkspaceModal extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      shareSuggestions: [],
-      groups: [],
-      originalAcl: [],
-      searchValue: '',
-      acl: [],
-      loaded: false
-    }
+const ShareWorkspaceModal = ({ onDismiss, workspace, workspace: { workspace: { namespace, name } } }) => {
+  // State
+  const [shareSuggestions, setShareSuggestions] = useState([])
+  const [groups, setGroups] = useState([])
+  const [originalAcl, setOriginalAcl] = useState([])
+  const [searchValue, setSearchValue] = useState('')
+  const [acl, setAcl] = useState([])
+  const [loaded, setLoaded] = useState(false)
+  const [working, setWorking] = useState(false)
+  const [updateError, setUpdateError] = useState(undefined)
+  const [lastAddedEmail, setLastAddedEmail] = useState(undefined)
 
-    this.list = createRef()
-  }
+  const list = useRef()
```

Comment:
Is this proper form to name a variable the same thing as a data-type?

@Shakespeared

Reply:
I could have called it, like, `listDomNode` or something, I just kept the name the same as it was, tho.

Reply:
oh yeah sorry should have clarified this wasn't a comment about a change due to this PR but was just wondering if in making this conversion if it would be appropriate to rename this variable (assuming `list` isn't a proper/standard variable name since it is the name of a data-type)